Expose the `batch` operator underlying rebuild

dominiklohmann opened the pull request:
This takes the internal rebatching operator of the rebuild command and makes it available as the `batch <limit>` operator. This is particularly useful for formats like Feather or Parquet or connectors like Kafka.

Additionally, this resolves a long-standing todo from the `sort` operator: We now rebatch its outputs transparently. In very much non-scientific measurements this increased the throughput of later operators in the pipeline significantly.

Note: to avoid merge conflicts this is based on #3390. I did want this to be reviewed separately, though.

mavam commented on libtenzir/builtins/operators/batch.cpp:
```diff
@@ -0,0 +1,108 @@
+    -> generator<table_slice> {
+    // TODO: This operator can massively benefit from an unordered
+    // implementation, where it can keep multiple buffers per schema. We should
+    // implement that once the ordering optimization has landed.
```
Do we have a story for that?

dominiklohmann replied:
I added something to our board.

mavam commented on libtenzir/builtins/operators/batch.cpp:
```diff
@@ -0,0 +1,108 @@
+    parser.add(limit, "<limit>");
+    parser.parse(p);
+    if (limit and limit->inner == 0) {
+      diagnostic::error("batch size must not be 0")
```
You could add `.primary(limit->source)` for better readability.

jachris replied:
I think ``"from `xyz`"`` is an anti-pattern in general. We only use this because we currently do not get the source location of the operator name. I'm hoping to change this soon.

mavam commented on web/docs/operators/transformations/batch.md:
```diff
@@ -0,0 +1,34 @@
+# batch
+
+The `batch` operator makes it possible to control batching precisely.
```
```suggestion
The `batch` operator controls the batch size of events.
```

mavam commented on web/docs/operators/transformations/batch.md:
````diff
@@ -0,0 +1,34 @@
+Write exactly 1 NDJSON object at a time to a Kafka topic.
+
+```
+... | batch 1 | to kafka -t topic write json -c
````
```suggestion
batch 1 | to kafka -t topic write json -c
```

We typically start with the very operator.

mavam reviewed (approved):
Nice.

Please add a few unit tests, as this is a lower-level building block.

dominiklohmann commented:
> Please add a few unit tests, as this is a lower-level building block.

I incorporated all the review feedback (thanks!) and added a few basic tests. I'll merge this into #3390 now to cut down on the number of CI runs I need to wait for as that is also already approved and the base of this.
